packages/lodestar/src/chain/chain.ts

```diff
@@ -166,12 +166,13 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     }
 
     await this.processBlock(block, blockHash);
-    const nextBlockInQueue = this.blockProcessingQueue.peek();
+    let nextBlockInQueue = this.blockProcessingQueue.peek();
     while (nextBlockInQueue) {
       const latestBlock = await this.db.block.getChainHead();
       if (nextBlockInQueue.parentRoot.equals(signingRoot(latestBlock, this.config.types.BeaconBlock))) {
         await this.processBlock(nextBlockInQueue, signingRoot(nextBlockInQueue, this.config.types.BeaconBlock));
         this.blockProcessingQueue.poll();
+        nextBlockInQueue = this.blockProcessingQueue.peek();
```

Comment:
whats the difference between poll and peek?

Reply:
Poll is return value and remove from queue. Peek is just return value

Reply:
should you just do poll before while loop and at this line and drop peek?

Reply:
I'm afraid of process block failing. But I guess if process blocks fail, then that means the block is probably bad and we should drop anyways... Hmm...